Standardizing Duckdb connector documentation (#608)
spiceai/docs: spiceaidocs/docs/components/data-connectors/duckdb.md

description: 'DuckDB Data Connector Documentation'
---

> DuckDB is an in-process SQL OLAP database management system designed for analytical query workloads. It is optimized for fast execution and can be embedded directly into applications, providing efficient data processing without the need for a separate database server.

Reviewer: What is OLAP?

Reviewer 2: Added in the newest version: Online Analytical Processing

> DuckDB is an in-process SQL OLAP database management system designed for analytical query workloads. It is optimized for fast execution and can be embedded directly into applications, providing efficient data processing without the need for a separate database server.

Reviewer 2: ```suggestion
DuckDB is an in-process SQL OLAP (Online Analytical Processing) database management system designed for analytical query workloads. It is optimized for fast execution and can be embedded directly into applications, providing efficient data processing without the need for a separate database server.
```

> This connector allows DuckDB [persistent database](https://duckdb.org/docs/connect/overview#persistent-database) to be used as a data source for federated SQL queries.

Reviewer 3: ```suggestion
This connector allows DuckDB [persistent databases](https://duckdb.org/docs/connect/overview#persistent-database) to be used as a data source for federated/accelerated SQL queries.
```

> This connector allows DuckDB [persistent databases](https://duckdb.org/docs/connect/overview#persistent-database) to be used as a data source for federated/accelerated SQL queries.

Reviewer: ```suggestion
This connector supports DuckDB [persistent databases](https://duckdb.org/docs/connect/overview#persistent-database) as a data source for federated SQL queries.
```

## Configuration

### `from`

The `from` field supports one of two forms:

| `from`                         | Description                                                                                                                                             |
| ------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `duckdb:database.schema.table` | Read data from a table named `database.schema.table` in the DuckDB file                                                                                |
| `duckdb:read_*()`              | Read data using one of the common [data import](https://duckdb.org/docs/data/overview) DuckDB functions, e.g `read_json`, `read_parquet` or `read_csv`. |

> 

Reviewer 2: DuckDB functions are not restricted to read_*(), functions like glob('*') can also be passed in, e.g.
```YAML
  - from: duckdb:glob('*')
    name: hits
```

Reviewer 3: Yes, any DuckDB functions that create a table/result set can be used

Author: Corrected in the latest push!

Author: Also I learned something new about the product

### `name`

The dataset name. This will be used as the table name within Spice.

Example:
```yaml
datasets:
  - from: duckdb:database.schema.table
    name: cool_dataset
    params:
      ...
```

```sql
SELECT COUNT(*) FROM cool_dataset;
```

```shell
+----------+
| count(*) |
+----------+
| 6001215  |
+----------+
```

### `params`

The DuckDB data connector can be configured by providing the following `params`:

| Parameter Name | Description                                        |
| -------------- | -------------------------------------------------- |

> | `duckdb_open`  | The name for the file to back the DuckDB database. |

Reviewer: ```suggestion
| `duckdb_open`  | The name of the DuckDB database to open. |
```

-- As a DuckDB function
SELECT * FROM read_json('todos.json');
```

> ## Secrets

Reviewer 3: Does the DuckDB data connector need any secrets?
